#167749938 Setup postgresql to replace mongoose database

ldonjibson opened the pull request:
#### What does this PR do?
Setup app to use postgresql database to persist data.

#### Description of Task to be completed?
Replace all instance of mongoose database in all reference files

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
App was previously setup with mongoose 

#### What are the relevant pivotal tracker stories?
[#167749938
](https://www.pivotaltracker.com/story/show/167749938)

daviddamilola commented:
nicely done

hustlaviola commented on src/database/config.js:
```diff
@@ -0,0 +1,12 @@
+import dotenv from 'dotenv';
+
+dotenv.config();
+const connetionstring = process.env.DATABASE_URL;
```
It's important to have a separate database for testing, development and production phases. We also need to follow the naming convention when creating variables 
```suggestion
const dbConnection = () => {
  const { NODE_ENV } = process.env;
  let dbURL;
  if (NODE_ENV === 'test') {
    dbURL = process.env.DATABASE_TEST_URL;
  } else if (NODE_ENV === 'development') {
    dbURL = process.env.DATABASE_DEV_URL;
  } else {
    dbURL = process.env.DATABASE_URL;
  }
  return dbURL;
};
const connectionString = dbConnection();
```

ldonjibson replied:
Alright

hustlaviola commented on src/database/config.js:
```diff
@@ -0,0 +1,12 @@
+const connetionstring = process.env.DATABASE_URL;
+
+const pool = new Pool({
+  connectionString: connetionstring,
```
```suggestion
  connectionString,
```

ldonjibson replied:
Thnaks

hustlaviola commented on .envSample:
```diff
@@ -0,0 +1,2 @@
+
```
This file should be renamed to `.env` on your local repo. It will contain the URL to your test and development databases. We do not need to merge this file, `.env` has been added to `.gitignore` already

ldonjibson replied:
I already a .env in my local environment.
it is just sample as i have named to show whoever is cloning how to configure the database.

hustlaviola commented on package.json:
```diff
@@ -20,10 +20,9 @@
-    "mongoose-unique-validator": "^2.0.1",
     "passport": "^0.4.0",
     "passport-local": "^1.0.0",
+    "pg": "^7.12.1",
```
Good job

hustlaviola commented on src/index.js:
```diff
@@ -8,7 +8,6 @@ const fs = require("fs"),
     cors = require("cors"),
     passport = require("passport"),
     errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
```
Good job on the removal of mongoose setup

ldonjibson replied:
thanks

hustlaviola reviewed (changes requested):
Great effort, you've done the most difficult task so far. However, we must implement some changes before merging.
Good job on PR comment. We can, however, do without the bullets when we are not making a list.

ldonjibson committed:
chore(postgresql):implement postgresql database

- replace mongoose database with postgresql
- add postgresql connection configuration
- remove all instances/references of mongoose database

[Maintains #167749938]
